test/e2e/gameserverallocation_test.go

```diff
@@ -69,6 +71,133 @@ func TestCreateFleetAndGameServerAllocate(t *testing.T) {
 	}
 }
 
+func TestCreateFleetAndGameServerStateFilterAllocation(t *testing.T) {
+	if !runtime.FeatureEnabled(runtime.FeatureStateAllocationFilter) {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	fleets := framework.AgonesClient.AgonesV1().Fleets(framework.Namespace)
+	fleet := defaultFleet(framework.Namespace)
+	ctx := context.Background()
+
+	flt, err := fleets.Create(ctx, fleet, metav1.CreateOptions{})
+	require.NoError(t, err)
+	defer fleets.Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	fleetSelector := metav1.LabelSelector{MatchLabels: map[string]string{agonesv1.FleetNameLabel: flt.ObjectMeta.Name}}
+	gsa := &allocationv1.GameServerAllocation{
+		Spec: allocationv1.GameServerAllocationSpec{
+			Required: allocationv1.GameServerSelector{LabelSelector: fleetSelector},
+		}}
+
+	// standard allocation
+	gsa, err = framework.AgonesClient.AllocationV1().GameServerAllocations(fleet.ObjectMeta.Namespace).Create(ctx, gsa, metav1.CreateOptions{})
```

Comment:
It's a bit strange to re-assign the variable `gsa` here right after defining it. Is there another name that would work to distinguish the above `gsa` from this one (in one place or the other)?

Reply:
🤔 curious - why do you think it's strange? If we aren't reusing the variable, doesn't seem like it would matter?

We do it a bunch in tests all over the place.

Reply:
I can't remember my rationale but I do remember being a bit confused looking at it. For some reason it seems more strange for allocations than for a pod / fleet / game server resource. But I think it's fine as-is.